### ISSUE TITLE
How to configure model relationships in case of not conventional naming db? Sorry I know this is not a code issue ...

### ISSUE THREAD

**Issue by etcware** [4spacesdk/CI4OrmExtension#21]
I'm not able to 'guess' model configuration params: I have an existing database that I cannot change and I need to define a many-to-many relation between ptv_section and ptv_navigation tables described in the following models. The many-to-many relation table is:

```
CREATE TABLE `ptv_navigation_sections` (
  `nav_sec_id` int(10) NOT NULL AUTO_INCREMENT,
  `navigation_fk` int(10) NOT NULL DEFAULT 0,
  `section_fk` int(10) NOT NULL DEFAULT 0,
   PRIMARY KEY (`nav_sec_id`),
  UNIQUE KEY `my_unique_index_pns` (`navigation_fk`,`section_fk`),
  KEY `navigation_fk` (`navigation_fk`) USING BTREE,
  KEY `section_fk` (`section_fk`) USING BTREE,
  CONSTRAINT `navigation_fk1` FOREIGN KEY (`navigation_fk`) REFERENCES `ptv_navigation` (`navigation_id`) ON DELETE CASCADE ON UPDATE NO ACTION,
  CONSTRAINT `section_fk2` FOREIGN KEY (`section_fk`) REFERENCES `ptv_sections` (`section_id`) ON DELETE CASCADE ON UPDATE NO ACTION
) ENGINE=InnoDB AUTO_INCREMENT=2653 DEFAULT CHARSET=utf8
```

SectionModel

```
...
        protected $table = 'ptv_sections';
	protected $primaryKey = 'section_id';
	public $hasMany = [
	    'nav' => [
	        'class' => NavigationModel::class,
	        'joinTable' => 'ptv_navigation_sections',
	        'otherField' => 'sec',
	        'joinSelfAs' => 'section_fk',
	        'joinOtherAs' => 'navigation_fk'
	    ],
	];
	public function getTableName() {
	    return $this->table;
	}
...
```

NavigationModel:

```
        protected $table = 'ptv_navigation';
	protected $primaryKey = 'navigation_id';
	
	public $hasMany = [
	    'sec' => [
	        'class' => SectionModel::class,
	        'joinTable' => 'ptv_navigation_sections',
	        'otherField' => 'nav',
	        'joinSelfAs' => 'navigation_fk',
	        'joinOtherAs' => 'section_fk'
	    ],
	];
	public function getTableName() {
	    return $this->table;
	}
```

**Comment by Martin-4Spaces** [comment]
It seems correct to me. Do you encounter any errors? And if so, how do I reproduce?

**Comment by etcware** [comment]
PPS I have the same relationship type with users and groups (in this case I followed the naming convention and the relationship table is groups_users) and the resulting query of the corresponding code is in the following:

```
foreach ($page_users as $user) {
    	    $user->groups = $user->groups->_getModel()->find();
    	    dd($user->groups->_getModel()->getLastQuery()->getQuery());
    	}
```
```
SELECT `groups`.*
FROM `groups`
LEFT OUTER JOIN `groups_users` `groups_users` ON `groups`.`id` = `groups_users`.`group_id`
LEFT OUTER JOIN `users` `users` ON `users`.`id` = `groups_users`.`user_id`
WHERE `users`.`id` = 1
```

**Comment by Martin-4Spaces** [comment]
This is a tricky one. You did everything correct. The issue is that OrmExtensiion has some assumptions about naming when doing join tables like this. I fixed that. Please try updating to latest dev-master version. And let me know if it works.
https://github.com/4spacesdk/CI4OrmExtension/commit/190857475663e9744aca61a6f3664baa70cb0792

**Comment by Martin-4Spaces** [comment]
You can find a working example of your scenario here https://github.com/4spacesdk/CI4-RestOrmExtensionSample/commit/418bfb9820d45353a3a582b7d6f0fb84515591be.

**Comment by etcware** [comment]
It works: too easy! :)

**Comment by etcware** [comment]
it works: too easy!

**Comment by etcware** [comment]
This example is working, my code no :( anyway I close this and I look for my error. Thanks

**Comment by etcware** [comment]
I'm reopening issue because I found the problem. I compared my software and the example you provided: my software didn't work and the example yes. I tried to reinstall CI4OrmExtension on the example:
```
composer remove 4spacesdk/ci4ormextension
composer require 4spacesdk/ci4ormextension
```
and I receive the same error of my software. The issue is in the  "4spacesdk/ci4ormextension": "^1.0" version, while the "4spacesdk/ci4ormextension": "dev-master" works!
